homeassistant/components/youtube/__init__.py

```diff
@@ -52,3 +56,17 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     if unload_ok := await hass.config_entries.async_unload_platforms(entry, PLATFORMS):
         hass.data[DOMAIN].pop(entry.entry_id)
     return unload_ok
+
+
+async def delete_devices(
+    hass: HomeAssistant, entry: ConfigEntry, coordinator: YouTubeDataUpdateCoordinator
+) -> None:
+    """Delete all devices created by integration."""
+    channel_ids = list(coordinator.data)
+    device_registry = dr.async_get(hass)
+    dev_entries = dr.async_entries_for_config_entry(device_registry, entry.entry_id)
+    for dev_entry in dev_entries:
+        if any(identifier[1] in channel_ids for identifier in dev_entry.identifiers):
+            device_registry.async_update_device(
```

Comment:
Shouldn't we add a test for this?

Reply:
I want to do a rewrite of the tests soon, but I wanted to get this in first